ssg/build_yaml.py

```diff
@@ -335,64 +326,141 @@ def __sub__(self, other):
+            for varname, value in to_expand.variables.items():
+                if varname not in self.variables:
+                    self.variables[varname] = value
+
+    def extend_by(self, extended_profile):
+        self.policies.update(extended_profile.policies)
+        super(ProfileWithSeparatePolicies, self).extend_by(extended_profile)
+
+
+class ProfileWithInlinePolicies(Profile):
+    def __init__(self, * args, ** kwargs):
+        super(ProfileWithInlinePolicies, self).__init__(* args, ** kwargs)
+        self.controls_by_policy = defaultdict(list)
+
+    def apply_selection(self, item):
+        if ":" in item:
+            policy_id, control_id = item.split(":", 1)
+            self.controls_by_policy[policy_id].append(control_id)
+        else:
+            super(ProfileWithInlinePolicies, self).apply_selection(item)
+
+    def resolve_controls(self, controls_manager):
+        for policy_id, controls in self.controls_by_policy.items():
+            if "all" in controls:
+                to_expand = self._policy_to_items(controls_manager, policy_id)
+            else:
+                to_expand = self._controls_to_items(controls_manager, policy_id, controls)
+
+            self.selected.extend(to_expand.rules)
+            for varname, value in to_expand.variables.items():
+                if varname not in self.variables:
+                    self.variables[varname] = value
+
+
+class ResolvableProfile(Profile):
+    def __init__(self, * args, ** kwargs):
+        super(ResolvableProfile, self).__init__(* args, ** kwargs)
+        self.resolved = False
+        self.resolved_selections = set()
+
+    def _control_to_items(self, controls_manager, policy_id, control_id):
+        return controls_manager.get_control(policy_id, control_id)
+
+    def _controls_to_items(self, controls_manager, policy_id, control_id_list):
+        items = Control()
+        for control_id in control_id_list:
+            new_control = self._control_to_items(controls_manager, policy_id, control_id)
+            items.rules.extend(new_control.rules)
+            items.variables.update(new_control.variables)
+        items.rules = list(set(items.rules))
+        return items
+
+    def _policy_to_items(self, controls_manager, policy_id):
+        control_id_list = controls_manager.get_all_controls(policy_id)
+        return self._controls_to_items(controls_manager, policy_id, control_id_list)
+
+    def resolve_controls(self, controls_manager):
+        pass
+
+    def extend_by(self, extended_profile):
+        extended_selects = set(extended_profile.selected)
+        self.resolved_selections.update(extended_selects)
+
+        updated_variables = dict(extended_profile.variables)
+        updated_variables.update(self.variables)
+        self.variables = updated_variables
+
+        updated_refinements = dict(extended_profile.refine_rules)
+        updated_refinements.update(self.refine_rules)
+        self.refine_rules = updated_refinements
```

Comment:
`resolve` now does more than just handling the inheritance, so I have extracted the parent profile merge to a separate method.